runtime/resolvers/glob.go

```diff
@@ -0,0 +1,376 @@
+	hash, err := hashstructure.Hash(r.props, hashstructure.FormatV2, nil)
+	if err != nil {
+		panic(err)
+	}
+	return strconv.FormatUint(hash, 16)
+}
+
+func (r *globResolver) Refs() []*runtimev1.ResourceName {
+	return nil
+}
+
+func (r *globResolver) Validate(ctx context.Context) error {
+	return nil
+}
+
+func (r *globResolver) ResolveInteractive(ctx context.Context) (runtime.ResolverResult, error) {
+	h, release, err := r.runtime.AcquireHandle(ctx, r.instanceID, r.props.Connector)
+	if err != nil {
+		return nil, err
+	}
+	defer release()
+
+	store, ok := h.AsObjectStore()
+	if !ok {
+		return nil, fmt.Errorf("connector %q is not an object store", r.props.Connector)
+	}
+
+	listObjectsProps := map[string]any{"path": r.props.Path}
+	for k, v := range r.props.AdditionalProps {
+		listObjectsProps[k] = v
+	}
+
+	entries, err := store.ListObjects(ctx, listObjectsProps)
+	if err != nil {
+		return nil, err
+	}
+
+	var rows []map[string]any
+	switch r.props.Partition {
+	case globPartitionTypeUnspecified, globPartitionTypeFile:
+		rows = r.buildFilesResult(entries)
+	case globPartitionTypeDirectory:
+		rows = r.buildPartitionedResult(entries, false)
+	case globPartitionTypeHive:
+		rows = r.buildPartitionedResult(entries, true)
+	default:
+		return nil, fmt.Errorf("unknown glob partition type %q", r.props.Partition)
+	}
+
+	if r.props.TransformSQL != "" {
+		rows, err = r.transformResult(ctx, rows, r.props.TransformSQL)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return runtime.NewMapsResolverResult(rows, r.makeResultSchema(rows)), nil
+}
+
+func (r *globResolver) ResolveExport(ctx context.Context, w io.Writer, opts *runtime.ResolverExportOptions) error {
+	return errors.New("not implemented")
+}
+
+// buildUnpartitioned builds a result consisting of one row per file.
+// Each row is a map with the keys "path" and "updated_on".
+func (r *globResolver) buildFilesResult(entries []drivers.ObjectStoreEntry) []map[string]any {
+	rows := make([]map[string]any, 0, len(entries))
+	for _, entry := range entries {
+		if entry.IsDir {
+			continue
+		}
+
+		rows = append(rows, map[string]any{
+			"path":       entry.Path,
+			"updated_on": entry.UpdatedOn,
+		})
+	}
+	return rows
+}
+
+// hivePartitionRegex is a regex that matches Hive-style partition values in a path.
+var hivePartitionRegex = regexp.MustCompile(`/([^/\?]+)=([^/\n\?]*)`)
+
+// buildPartitioned builds a result consisting of one row per partition.
+// It groups the files by directory.
+// Each row is a map with the keys "path", "updated_on", "files", and the Hive partition columns if requested.
+func (r *globResolver) buildPartitionedResult(entries []drivers.ObjectStoreEntry, parseHivePartitions bool) []map[string]any {
+	// Group the entries by directory
+	rows := make(map[string]map[string]any)
+	for _, entry := range entries {
+		if entry.IsDir {
+			continue
+		}
+
+		dir := path.Dir(entry.Path)
+
+		row := rows[dir]
+		if row == nil {
+			// Init a new row
+			row = map[string]any{
```

Comment:
May be add a struct for `row` and convert it to `map[string]any` while returning to avoid type assertions in below code ?

Reply:
I actually did that initially, but the problem is that `parseHivePartitions` can add arbitrary keys